service/integration/registered_resources_test.go

```diff
@@ -0,0 +1,851 @@
+	s.Equal(values[1], createdVals[1].GetValue())
+}
+
+func (s *RegisteredResourcesSuite) Test_CreateRegisteredResource_WithMetadata_Succeeds() {
+	req := &registeredresources.CreateRegisteredResourceRequest{
+		Name: "test_create_res_metadata",
+		Metadata: &common.MetadataMutable{
+			Labels: map[string]string{
+				"key1": "value1",
+				"key2": "value2",
+			},
+		},
+	}
+
+	created, err := s.db.PolicyClient.CreateRegisteredResource(s.ctx, req)
+	s.Require().NoError(err)
+	s.NotNil(created)
+	s.Require().Len(created.GetMetadata().GetLabels(), 2)
+}
+
+func (s *RegisteredResourcesSuite) Test_CreateRegisteredResource_WithNonUniqueName_Fails() {
+	existing := s.f.GetRegisteredResourceKey("res_with_values")
+	req := &registeredresources.CreateRegisteredResourceRequest{
+		Name: existing.Name,
+	}
+
+	created, err := s.db.PolicyClient.CreateRegisteredResource(s.ctx, req)
+	s.Require().Error(err)
+	s.Require().ErrorIs(err, db.ErrUniqueConstraintViolation)
+	s.Nil(created)
+}
+
+// Get
+
+func (s *RegisteredResourcesSuite) Test_GetRegisteredResource_Succeeds() {
+	existingRes := s.f.GetRegisteredResourceKey("res_only")
+
+	got, err := s.db.PolicyClient.GetRegisteredResource(s.ctx, existingRes.ID)
+	s.Require().NoError(err)
+	s.NotNil(got)
+	s.Equal(existingRes.Name, got.GetName())
+	metadata := got.GetMetadata()
+	s.False(metadata.GetCreatedAt().AsTime().IsZero())
+	s.False(metadata.GetUpdatedAt().AsTime().IsZero())
+	s.Empty(got.GetValues())
+}
+
+func (s *RegisteredResourcesSuite) Test_GetRegisteredResource_WithValues_Succeeds() {
+	existingRes := s.f.GetRegisteredResourceKey("res_with_values")
+	existingResValue1 := s.f.GetRegisteredResourceValueKey("res_with_values__value1")
+	existingResValue2 := s.f.GetRegisteredResourceValueKey("res_with_values__value2")
+
+	got, err := s.db.PolicyClient.GetRegisteredResource(s.ctx, existingRes.ID)
+	s.Require().NoError(err)
+	s.NotNil(got)
+	values := got.GetValues()
+	s.Require().Len(values, 2)
+	s.Equal(existingResValue1.Value, values[0].GetValue())
+	s.Equal(existingResValue2.Value, values[1].GetValue())
+}
+
+func (s *RegisteredResourcesSuite) Test_GetRegisteredResource_InvalidID_Fails() {
+	got, err := s.db.PolicyClient.GetRegisteredResource(s.ctx, invalidID)
+	s.Require().Error(err)
+	s.Require().ErrorIs(err, db.ErrNotFound)
+	s.Nil(got)
+}
+
+// List
+
+func (s *RegisteredResourcesSuite) Test_ListRegisteredResources_NoPagination_Succeeds() {
+	existingRes := s.f.GetRegisteredResourceKey("res_with_values")
+	existingResValue1 := s.f.GetRegisteredResourceValueKey("res_with_values__value1")
+	existingResValue2 := s.f.GetRegisteredResourceValueKey("res_with_values__value2")
+	existingResOnly := s.f.GetRegisteredResourceKey("res_only")
+
+	list, err := s.db.PolicyClient.ListRegisteredResources(s.ctx, &registeredresources.ListRegisteredResourcesRequest{})
+	s.Require().NoError(err)
+	s.NotNil(list)
+
+	foundCount := 0
+
+	for _, r := range list.GetResources() {
+		if r.GetId() == existingRes.ID {
```

Comment:
Could this test find the same `existingRes` twice and pass incorrectly here?

Reply:
The `existingRes` and `existingResOnly` records would only be in the results list once each, so that shouldn't happen. The real issue though is that I'm relying on the order of the values retrieved, which is flaky since there is no strict ordering when building the JSON for the values. Going to refactor the GET/LIST tests.